cfme/containers/pod.py

```diff
@@ -38,12 +39,27 @@ def __init__(self, name, provider):
         self.name = name
         self.provider = provider
 
+    def _on_detail_page(self):
+        """ Returns ``True`` if on the providers detail page, ``False`` if not."""
+        ensure_browser_open()
+        return sel.is_displayed('//div//h1[contains(., "{} (Summary)")]'.format(self.name))
+
     def load_details(self, refresh=False):
         if not self._on_detail_page():
             self.navigate(detail=True)
         elif refresh:
             tb.refresh()
 
+    def get_detail(self, *ident):
+        """ Gets details from the details infoblock
+
+        Args:
+            *ident: An InfoBlock title, followed by the Key name, e.g. "Relationships", "Images"
+        Returns: A string representing the contents of the InfoBlock's value.
+        """
+        self.navigate(detail=True)
```

Comment:
This should rather do ``self.load_details(refresh=True)`` to not navigate the whole path on each query.